rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -49,6 +49,56 @@ SPECIALIZE_GENERIC_C_ARRAY(uint32, uint32_t)
 SPECIALIZE_GENERIC_C_ARRAY(int64, int64_t)
 SPECIALIZE_GENERIC_C_ARRAY(uint64, uint64_t)
 
+// TODO(mikaelarguedas) find right spot to define these
```

Comment:
I agree that making the existing work for this use case is a good idea.

I think it is ok to keep the new struct and functions where they are. If we ever need them somewhere else we can easily refactor / move them to a more generic place.